draft-ietf-tls-tls13.md

```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
 assigned by IANA in the TLS Content Type Registry as described in
 {{iana-considerations}}.
 
-Application Data messages are carried by the record layer and are
-fragmented and encrypted as described below. The messages are treated
-as transparent data to the record layer.
-
 ## Record Layer
 
 The TLS record layer receives uninterpreted data from higher layers in
 non-empty blocks of arbitrary size.
 
-The record layer fragments information blocks into TLSPlaintext records
-carrying data in chunks of 2^14 bytes or less. Message boundaries are
-not preserved in the record layer (i.e., multiple messages of the same
-ContentType MAY be coalesced into a single TLSPlaintext record, or a single
-message MAY be fragmented across several records).
-Alert messages ({{alert-protocol}}) MUST NOT be fragmented across records.
+The record layer fragments information blocks into TLSPlaintext
+records carrying data in chunks of 2^14 bytes or less. Message
+boundaries are handled differently depending on the underlying
+ContentType and MUST following rules.
+
+Alert messages ({{alert-protocol}}) MUST NOT be fragmented across
+records and multiple Alert messages MUST NOT be coalesced into a
+single TLSPlaintext record. In other words, a record with an Alert
+type MUST contain exactly one message. This is the default behaviour
+to be adopted by new record content type.
+
+Application Data messages are carried by the record layer and are
+fragmented and encrypted as described below. The messages are treated
+as transparent data to the record layer. Zero-length
+fragments of Application Data MAY be sent as they are potentially
+useful as a traffic analysis countermeasure.
+
+Since Handshake messages can be larger than a record content, a single
+Handshake message MAY be fragmented across several records. Moreover,
+multiple Handshake messages MAY be coalesced into a single
+TLSPlaintext record. However, the following constraints MUST be
+observed:
+
+- Implementations MUST NOT send zero-length fragments of Handshake
+  types, even if those fragments contain padding.
+
+- Handshake messages MUST NOT be interleaved with other record
+  types. That is, if a message is split over two or more handshake
+  records, there MUST NOT be any other records between them.
+
+RFC EDITOR: The following point could be completely moved from the 4.6
+section if needed. This way, all the relevant information concerning
+record splitting/coalescing/interleaving would be in the same place.
```

Comment:
Don't instruct the RFC editor here, just make the change you think best.  FWIW, it is good to repeat this here.